File: app/src/main/java/vn/tiki/android/androidhometest/mvp/presenter/DealContract.java
package vn.tiki.android.androidhometest.mvp.presenter;

import android.support.annotation.NonNull;

import java.util.List;
import vn.tiki.android.androidhometest.mvp.model.remote.api.response.Deal;
import vn.tiki.android.androidhometest.mvp.view.BaseView;


public class DealContract {

    public interface View extends BaseView<Presenter> {
        void showDealList(List<Deal> deals);
        boolean isActive();
        void showDealDetailUi(Deal deal);


    }

    public interface Presenter extends BasePresenter {
        void loadDealList(boolean forceUpdate);
        void openDealDetail(@NonNull Deal deal);
    }


}

File: app/src/main/java/vn/tiki/android/androidhometest/mvp/presenter/DealContract.kt
package vn.tiki.android.androidhometest.mvp.presenter

import vn.tiki.android.androidhometest.mvp.model.remote.api.response.Deal
import vn.tiki.android.androidhometest.mvp.view.BaseView


class DealContract {

    interface View : BaseView<Presenter> {
        val isActive: Boolean
        fun showDealList(deals: List<Deal>)
        fun showDealDetailUi(deal: Deal)


    }

    interface Presenter : BasePresenter {
        fun loadDealList(forceUpdate: Boolean)
        fun openDealDetail(deal: Deal)
    }


}
